### PR TITLE
Addresses #3, Adding basic tests for results component

### PR DESCRIPTION
**Pull request by Marauderer97**
Related to issue #3 

Changes: Adds basic tests for results component

Demo Link:
[Here](https://marauderer97.github.io/susper.com/)
Screenshots for the change: 

(Not applicable)

@nikhilrayaprolu @harshit98 Please check, I have also updated the sheet posted by @nikhilrayaprolu in Gitter(as many columns as I understood what to fill)

### CODE REVIEW

**Comment by nikhilrayaprolu** [comment]
@Marauderer97 everything looks good, please follow up with @harshit98 's suggestion.

**Comment by Marauderer97** [comment]
@harshit98 I wrote the basic ones to ensure all major components are present. Can you suggest any?

File: src/app/results/results.component.spec.ts [hunk 1]
```diff
@@ -93,4 +93,44 @@ describe('ResultsComponent', () => {
...
+  it('should have an app-theme element', () => {
+    let compiled = fixture.debugElement.nativeElement;
+
+    expect(compiled.querySelector('app-theme')).toBeTruthy();
+  });
+
+  it('should have a pagination bar', () => {
+    let compiled = fixture.debugElement.nativeElement;
+    expect(compiled.querySelector('div.pagination-bar'));
+  });
```
Comment (harshit98) [inline src/app/results/results.component.spec.ts]: Remove it. As we are going to implement a separate component for it. We don't need it right now.

**Comment by harshit98** [comment]
@Marauderer97 

1.  It should have tools as dropdown menu.
2.  It should display message: `About results of xyz`
3.  It should have app-navbar element too.
4.  For 'All' results it should be have, `item.title`, `item.link` and `item.description`
5.  It should have element `app-infobox`
6. Repeat 4 for image results and video results as well.
7. It should have app-advancedsearch element.
8. It should have app-footernavbar element.

**Comment by Marauderer97** [comment]
@harshit98 
aren't footer-navbar, navbar and advanced search already included, rest working on it

File: src/app/results/results.component.spec.ts (hunk 1 above)
Comment (nikhilrayaprolu) [inline src/app/results/results.component.spec.ts]: no need to remove this now.right tests for pagination bar too. once a separate component is implemented for pagination bar. we will be migrating tests related to pagination bar to its own component test files.

**Comment by harshit98** [comment]
Okay, @Marauderer97 I didn't notice that. You can work on rest which are not present.

**Comment by Marauderer97** [comment]
@harshit98 working on this, facing a block in the sense, in most tests I either need to set the value of some component variables and check. or use conditional statements, this is not passing ng test, I would appreciate any suggestions

**Comment by Marauderer97** [comment]
@harshit98 I have added all the tests you specified along with a few extra, please check now

File: src/app/results/results.component.spec.ts [hunk 2]
```diff
@@ -93,4 +93,77 @@ describe('ResultsComponent', () => {
...
+  it('should have an app-related-search element', () => {
+    let compiled = fixture.debugElement.nativeElement;
+
+    expect(compiled.querySelector('app-related-search')).toBeTruthy();
+  });
+
+  it('should have an app-advancedsearch element', () => {
+    let compiled = fixture.debugElement.nativeElement;
+
+    expect(compiled.querySelector('app-advancedsearch')).toBeTruthy();
```
Comment (harshit98) [inline src/app/results/results.component.spec.ts]: Remove it. It is getting repeated.

File: src/app/results/results.component.spec.ts [hunk 3]
```diff
@@ -93,4 +93,77 @@ describe('ResultsComponent', () => {
...
+
+
+  it('should have a search options menu', () => {
+    let compiled = fixture.debugElement.nativeElement;
+    expect(compiled.querySelector('div#search-options-field ul#search-options'));
+  });
+
+  it('should have "items$" variable', () => {
+
+    expect(component.items$).toBeTruthy();
```
Comment (harshit98) [inline src/app/results/results.component.spec.ts]: Remove extra line.

File: src/app/results/results.component.spec.ts [hunk 4]
```diff
@@ -93,4 +93,77 @@ describe('ResultsComponent', () => {
...
+  });
+
+
```
Comment (harshit98) [inline src/app/results/results.component.spec.ts]: Remove this extra line too.

**Comment by harshit98** [comment]
@Marauderer97 I have gone through the code again. Kindly do changes which I have suggested. :)
Also, test for `footer-navbar` element is not there. Can you add it, please? Other test case looks good to me. :+1:

**Comment by harshit98** [comment]
@Marauderer97 don't remove the test for pagination too. I'm sorry for causing inconvenience but @nikhilrayaprolu suggested those test will be migrated once a separate component for pagination will be implemented.

**Comment by Marauderer97** [comment]
@harshit98 Did all the requested changes please check now

**Comment by mariobehling** [comment]
Travis was failing. Please re-check.

**Comment by Marauderer97** [comment]
@harshit98 Done!

**Pull request merged**